### PR TITLE
adding pagination in audit table partition query

### PR DESCRIPTION
**Pull request by zhljen**

### CODE REVIEW

File: metacat-common/src/main/java/com/netflix/metacat/common/dto/GetPartitionsRequestDto.java [hunk 1]
```diff
@@ -31,4 +31,8 @@
     private String filter;
     private List<String> partitionNames;
     private Boolean includePartitionDetails = false;
+    private Boolean includeAuditOnly = false; //for audit tables with wap patterns
```
Comment (ajoymajumdar) [inline metacat-common/src/main/java/com/netflix/metacat/common/dto/GetPartitionsRequestDto.java]: Add the comment on the top of the field.

File: metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy [hunk 2]
```diff
@@ -511,19 +511,53 @@ class MetacatSmokeSpec extends Specification {
...
         then:
         noExceptionThrown()
-        assert partkeys.size() == 1
+        assert partkeys.size() == 2
+
         assert partkeys.get(0).equals("one=1")
+
+        assert auditparts.size() == autoPartSize
+
+
```
Comment (ajoymajumdar) [inline metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy]: add cleanup: step to drop the table.

File: metacat-main/src/main/java/com/netflix/metacat/main/services/impl/MViewServiceImpl.java [hunk 3]
```diff
@@ -288,7 +288,7 @@ public PartitionsSaveResponseDto savePartitions(
             final List<String> partitionNames = partitionDtos.stream().map(
                 partitionDto -> partitionDto.getName().getPartitionName()).collect(Collectors.toList());
             final List<PartitionDto> existingPartitions = partitionService
-                .list(viewQName, null, partitionNames, null, null, false, false, false);
+                .list(viewQName, null, partitionNames, null, null, false, false, false, false);
```
Comment (ajoymajumdar) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/impl/MViewServiceImpl.java]: Shoudnt the includeAuditOnly field be true?

File: metacat-main/src/main/java/com/netflix/metacat/main/services/impl/MViewServiceImpl.java [hunk 4]
```diff
@@ -375,7 +375,7 @@ public void deletePartitions(final QualifiedName name, final List<String> partit
             QualifiedName.ofTable(name.getCatalogName(), VIEW_DB_NAME, createViewName(name));
         return partitionService
             .list(viewQName, filter, partitionNames, sort, pageable, includeUserMetadata, includeUserMetadata,
-                includePartitionDetails);
+                includePartitionDetails, false);
```
Comment (ajoymajumdar) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/impl/MViewServiceImpl.java]: includeAuditOnly should be true?

**Comment by zhljen** [comment]
Added changes to address comments.

**Pull request merged**